js/dfpslotsprovider.js

```diff
@@ -214,7 +219,11 @@ export default class DFPSlotsProvider extends React.Component {
   }
 
   render() {
-    const children = <div> {this.props.children} </div>;
+    let children = <div> {this.props.children} </div>;
+    if (React.Fragment) {
```

Comment:
@scott-thrillist I'm really sorry to bother you about this again :-(, I really-really appreciate your contribution and I don't want to be a pita)  I'm trying to see react-dfp behavior from the point of the client apps, having react-dfp outputting different dom structure/behavior depending of the react version seems a bit confusing to me (and error prone, think in scenarios like debugging, styling, etc), don't you agree?

Do you really need this?, if that's the case i would really love to know why :-)

Reply:
IDK about OP, but for me, it would be great to have fragments there so that SSR hydration does not break if the context is not present. 

ATM I am injecting the context into the SSR jsx, but I'd rather not do it. So it's not a big issue per se.

WDYT about adding an optional parameter called something like `childrenWrapper` that defaults to Fragment? This would allow the legacy users to pass div or any component? Or, in order to not create a new major version, make it vice versa and allow passing the fragment with div being the default?

Reply:
> IDK about OP, but for me, it would be great to have fragments there so that SSR hydration does not break if the context is not present.

As far I am aware, react-dfp should support SSR without issues, or at least it used to be the case... I wonder if this is a new bug introduced with the last updates I pushed to support the new context api.

@mmaxacov, could u please share a code snippet showing how your code looks? and also please tell me what version of react you are using.... i'll try to reproduce the bug at my side.

thanks

Reply:
@jaanauati we have a separate file for SSR that does not wrap the app with the provider in order to minimize the server execution time. It builds the HTML and sends in to the client. On the client, the app is being rendered again with HTML, but now there is the provider present. Since the provider wraps the whole app in a div, the hidration fails, since there is now one extra div at the top of the hierarchy.

It's not a big issue in general, so feel free to ignore it, but in my personal opinion, providers are an ethereal, logical abstraction and them rendering HTML looks spooky.

Reply:
@mmaxacov 
> It's not a big issue in general, so feel free to ignore it, but in my personal opinion, providers are an ethereal, logical abstraction and them rendering HTML looks spooky.


I see your point now, thank you for the explanation!, I totally agree w you on this.

According @apascal the `<div>` wrapper may not be needed at all: 
https://github.com/jaanauati/react-dfp/pull/164

@scott-thrillist could u please revert this block :-) ? the discussion will continue in the other pull request. 

thanks!

Reply:
This is done